src/components/ReviewConfirm/IncorporationDateTime.vue

```diff
@@ -204,65 +234,66 @@ export default class IncorporationDateTime extends Mixins(DateMixin) {
       const dateToParse = effectiveDate && new Date(effectiveDate)
 
       if (dateToParse) {
-        // Parse the Date
-        const year = dateToParse.getFullYear()
-        const month = dateToParse.getMonth()
-        const day = dateToParse.getDate()
-
         // Parse the Time and display DateTime
-        if (year && month && day) {
-          let hour = dateToParse.getHours()
-          const minute = dateToParse.getMinutes()
-          const amPm = hour >= 12 ? 'PM' : 'AM'
-
-          if (hour > 12) {
-            hour -= 12
-          } else if (hour === 0) {
-            hour = 12
-          }
-
-          this.dateText = `${year}-${month + 1}-${day}`
-          this.selectHour = [hour.toLocaleString()]
-          this.selectMinute = [minute.toLocaleString('en-US', { minimumIntegerDigits: 2 })]
-          this.selectPeriod = amPm
+        let hour = dateToParse.getHours()
+        const minute = dateToParse.getMinutes()
+        const amPm = hour >= 12 ? 'PM' : 'AM'
+
+        if (hour > 12) {
+          hour -= 12
+        } else if (hour === 0) {
+          hour = 12
         }
-      }
-    }
-  }
-
-  /** Validate the DateTime is within the allowed range */
-  private isValidDateTime (): boolean {
-    if (this.incorporationDateTime.effectiveDate) {
-      const effectiveDate = this.incorporationDateTime.effectiveDate
-      const dateToParse = new Date(effectiveDate)
-      const startDate = new Date()
-
-      const timeDiff = dateToParse.getTime() - startDate.getTime()
-      const timeDiffInMinutes = Math.floor(timeDiff / 1000 / 60)
 
-      // Time set must be more than 2 minutes and less than 10 days
-      return timeDiffInMinutes >= 2 && timeDiffInMinutes <= 14400
+        this.dateText = this.dateToUsableString(dateToParse)
+        this.selectHour = [hour.toLocaleString()]
+        this.selectMinute = [minute.toLocaleString('en-US', { minimumIntegerDigits: 2 })]
+        this.selectPeriod = amPm
+      }
     }
-    return false
   }
 
-  /** The maximum date that can be entered. */
+  /** The minimum date that can be entered. */
   private get minDate (): string {
     const minDate = new Date()
 
-    return this.dateToUsableString(minDate)
+    return minDate.toISOString()
   }
 
   /** The maximum date that can be entered. */
   private get maxDate (): string {
     const maxDate = new Date()
     maxDate.setDate(new Date().getDate() + 10)
 
-    return this.dateToUsableString(maxDate)
+    return maxDate.toISOString()
+  }
+
+  /** The minimum time that can be entered. */
+  private get minTime (): string {
+    return new Date(new Date().getTime() + 180000)
+      .toLocaleTimeString('en-US', { hour: '2-digit', minute: '2-digit', hour12: true })
+  }
+
+  /** The maximum time that can be entered. */
+  private get maxTime (): string {
+    const maxDate = new Date(new Date().setDate(new Date().getDate() + 10))
```

Comment:
Are these many new Date() needed?

Reply:
Out of interest, set a break point (or add a console.log) to this method to see how often it's called. If it's called a lot, I'd be tempted to optimize it.

Reply:
Each Date.prototype method used here returns a number ( one in days and one in milliseconds ). Need to then convert that number back into an actual Date. I could write a reference tho `const date = new Date()`?

Reply:
I will check, :re the call amounts!

Reply:
let maxDate = new Date ()
maxDate.setDate(maxDate.getDate() + 10) or something like this

Reply:
absolutely, can do!

Reply:
Mentioned this in a previous comment, but `Date.now()` may help if you are interested in the milliseconds out the gate without creating a new date object. A little bit more efficient, especially if there are certain functions that get evaluated multiple times for things like validation.

Reply:
Updated!